core/src/main/java/hudson/model/Actionable.java

```diff
@@ -91,18 +95,51 @@
      */
     @Exported(name="actions")
     public final List<? extends Action> getAllActions() {
-        List<Action> _actions = new ArrayList<Action>(getActions());
-        for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
-            if (taf.type().isInstance(this)) {
-                try {
-                    _actions.addAll(createFor(taf));
-                } catch (Exception e) {
-                    LOGGER.log(Level.SEVERE, "Could not load actions from " + taf + " for " + this, e);
+        List<Action> _actions = getActions();
+        boolean adding = false;
+        synchronized (Actionable.class) {
```

Comment:
Maybe, maybe not - explicit contention for something that will be invoked often by numerous threads raises a flag for me on general principles.  I agree it *shouldn't* be as expensive as initially I thought when skimming through (and hopefully won't be a problem), but still worth a 🐜 on general principles.  

> Any kind of cache lookup needs to acquire a lock anyway, so we might as well use just one

IIRC most of the hashmaps and concurrent maps are based on locking per-slot and not on the whole, so contention is extremely rare.  Actions are likely to get requested by many threads, and frequently, so contention will be high, even if very brief.